### PR TITLE
Check hg hssp

### PR DESCRIPTION
**Pull request by cbaakman**
get_hg_hssp is now no longer a task, but a subroutine of mkhssp_from_sequence.

Whenever a stockholm file is requested for a sequence, files in the hg-hssp directory might be used.

### CODE REVIEW

File: xssp_rest/tasks.py [hunk 1]
```diff
@@ -252,10 +260,9 @@ def get_task(input_type, output_type):
         if output_type == 'hssp_hssp' or \
            output_type == 'hssp_stockholm':
```
Comment (jonblack) [inline xssp_rest/tasks.py]: Doesn't the type `hg_hssp` need to be added here? Have you tested this by running it in the browser?

File: xssp_rest/tasks.py [hunk 2]
```diff
@@ -179,6 +186,7 @@ class NULL_NAMESPACE:
     path = os.path.join(flask_app.config['HG_HSSP_ROOT'],
                         sequence_hash + '.sto.bz2')
 
+    _log.debug("Checking for existence of {}".format(path))
```
Comment (jonblack) [inline xssp_rest/tasks.py]: This debug statement doesn't add any value.

File: xssp_rest/tasks.py [hunk 3]
```diff
@@ -94,6 +94,14 @@ def mkhssp_from_sequence(sequence, output_format):
 
     If present, the input FASTA description line is used.
     """
+
+    if output_format == 'hssp_stockholm':
+        try:
+            output = get_hg_hssp(sequence)
+            return output
+        except RuntimeError as e:
+            _log.info("hg-hssp content cannot be acquired, continuing without")
```
Comment (jonblack) [inline xssp_rest/tasks.py]: The log message should say something like "No cached data found. Data will be generated now.".

File: xssp_rest/tasks.py [hunk 4]
```diff
@@ -94,6 +94,14 @@ def mkhssp_from_sequence(sequence, output_format):
 
     If present, the input FASTA description line is used.
     """
+
+    if output_format == 'hssp_stockholm':
```
Comment (jonblack) [inline xssp_rest/tasks.py]: Why `hssp_stockholm`? The user interface in the browser has these options. Are you asserting that `hssp_stockholm` and `hg_hssp` are the same?

File: tests/unit/test_tasks.py [hunk 5]
```diff
@@ -282,18 +282,20 @@ def test_get_hg_hssp(self, mock_exists, mock_bz2file):
...
-    @raises(RuntimeError)
-    def test_get_hg_hssp_file_not_found(self, mock_exists):
-        from xssp_rest.tasks import get_hg_hssp
-        get_hg_hssp('TTCCPSIVARSNFNVCRLPGTPEAICATYTGCIIIPGATCPGDYAN')
+#    @patch('os.path.exists', return_value=False)
+#    @raises(RuntimeError)
+#    def test_get_hg_hssp_file_not_found(self, mock_exists):
+#        from xssp_rest.tasks import mkhssp_from_sequence
+#        mkhssp_from_sequence('TTCCPSIVARSNFNVCRLPGTPEAICATYTGCIIIPGATCPGDYAN',
+#                             'stockholm')
```
Comment (jonblack) [inline tests/unit/test_tasks.py]: Remove commented out code.

File: tests/unit/test_tasks.py [hunk 6]
```diff
@@ -330,8 +332,8 @@ def test_get_task(self):
         eq_(task.__name__, 'mkhssp_from_sequence')
         task = get_task('sequence', 'hssp_stockholm')
         eq_(task.__name__, 'mkhssp_from_sequence')
-        task = get_task('sequence', 'hg_hssp')
-        eq_(task.__name__, 'get_hg_hssp')
+#        task = get_task('sequence', 'hg_hssp')
+#        eq_(task.__name__, 'get_hg_hssp')
```
Comment (jonblack) [inline tests/unit/test_tasks.py]: Remove commented out code.

File: tests/unit/test_tasks.py [hunk 7]
```diff
@@ -282,18 +282,20 @@ def test_get_hg_hssp(self, mock_exists, mock_bz2file):
         instance = mock_bz2file.return_value
         instance.__enter__.return_value.read.return_value = 'data'
 
-        from xssp_rest.tasks import get_hg_hssp
-        content = get_hg_hssp('TTCCPSIVARSNFNVCRLPGTPEAICATYTGCIIIPGATCPGDYAN')
-
+        from xssp_rest.tasks import mkhssp_from_sequence
+        result = mkhssp_from_sequence.delay('TTCCPSIVARSNFNVCRLPGTPEAICATYTGCIIIPGATCPGDYAN',
+                                            'hssp_stockholm')
```
Comment (jonblack) [inline tests/unit/test_tasks.py]: The test is for `hg_hssp` but the type is `hssp_stockholm`. See my other comments.

File: tests/integration/test_xssp_rest.py [hunk 8]
```diff
@@ -85,6 +85,20 @@ def _test_input_ids(self, input_type, output_types, test_ids):
...
+
+        The test tries to get a known existing HG-HSSP file
+        using the XSSP REST API.
+        """
+        test_seqs = ['MLAGAGRPGLPQGRHLCWLLCAFTLKLCQAEAPVQEEKLSASTSNLPCWLVEE' +
+        'FVVAEECSPCSNFRAKTTPECGPTGYVEKITCSSSKRNEFKSCRSALMEQRLFWKFEGAVVCVALI' +
+        'FACLVIIRQRQLDRKALEKVRKQIESI']
+        _log.info('Testing an existing hssp input: {}'.format(
+            ' '.join(test_seqs)))
+        self._test_input_ids('sequence', ['hssp_stockholm'],
```
Comment (jonblack) [inline tests/integration/test_xssp_rest.py]: See my other comments about `hg_hssp` and `hssp_stockholm`.

File: xssp_rest/tasks.py (hunk 4 above)
Comment (cbaakman) [inline xssp_rest/tasks.py]: the format of hg-hssp files is stockholm, so before returning anything the service has to check that this is what the user wants.
Optionally, we could do a conversion if the user wants hssp_hssp instead.

**Comment by jonblack** [comment]
It looks like this isn't the best solution. xssp-rest doesn't cache results, so the result will be lost. It's better if the databanks provide an interface for this where the result can be saved along with the rest of the generated data. It's not the responsibility of xssp-rest to do this. This also applies to the creation of other hssp files that's already implemented, but that wasn't considered at the time. For now we'll look at improving the databank services.

**Pull request closed without merging**